README.md

```diff
@@ -21,14 +21,56 @@ Once you have implemented the `fm` program (and tested it to the best of your ab
 
 *What preprocessing data do you store in files, and how?*
 
+We store the preprocessed data in a .txt file with \n between each data point. For multiple sequnences in the same fasta file, one is stored after the other. We store the following:
+* The sequence header. This is needed for the output.
+* n, the length of the sequence. Can be deduced from the other data, but make sense to save, since we can then malloc with it. 
+* σ, alphabetsize. 
+* bwt of the converted sequence (where the chars are ints 0...σ-1). With commas between each char. We could same O and C, but they would take more space. 
+* SA. Comma seperated
+* The alphabet. No seperation chars, since each symbol is a unique char. 
+
+
 ## Insights you may have had while implementing the algorithm
+It takes 90% of the time to do the last 10% of the code, testing and writting the report. 
+
+Between this and the last handin, it has become part of the coding process to do most of the coding on one (Windows) machine, push the code and finish it on another (Linux). The code will be running fine on the first machine, but will be buggy on the other, which we will then fix. It is unclear which types of bugs is ignored, but it seems like the first machine is more forgiving with memory. 
+
+There are many ways to run the code. We mainly use the 'run' and 'debug' buttons in the editor (CLion) or run the code from the terminal. We have learned that the code will sometimes fail in one, but not the others. This is expecially weird when there is an error when running the code in the editor, but not when debugging. These erros seem to be the same we find by switching machines. 
```

Comment:
It might be that there is uninitialized memory that sits elsewhere with or without the debugger, or maybe the editor compiles with different flags or something. It happens quite often, to be honest.

Reply:
So when we face this issue, it is most likely because "we were lucky" when we execute from the terminal - correct?

We had some issues when we were retrieving something out-of-bounds ... probably because of off-by-one errors. However, they were hard to detect: Wouldn't the OS send a SIGSEGV if we go out-of-bounds? Maybe we allocated too much memory, didn't get a SIGSEGV, and thus resulting in reading "random" stuff from memory?

Reply:
Ah; so for instance if we malloc something of size x, and the OS gives us a page of size x+k (for k >= 0), we can then read from memory at location x+1, x+2, .. x+k without getting a segfault?

Reply:
As a rule of thumb, yes. The OS only checks if you have access to memory in the resolution the hardware gives it, which is usually pages of a kilo or mega byte.

Reply:
Thanks! That makes much more sense now.